src/reducers/ui.js

```diff
@@ -51,11 +74,13 @@ export default function ui(stateIn, action) {
       return state.set('workspace', DEFAULT_WORKSPACE);
 
     case 'HIDE_COMPONENT':
+      return restoreFlexOnComponentToggle(action.payload.componentName, state);
+
     case 'UNHIDE_COMPONENT':
-      if (action.payload.componentName === 'output') {
-        return state.setIn(['workspace', 'rowFlex'], DEFAULT_ROW_FLEX);
-      }
-      return state.setIn(['workspace', 'columnFlex'], DEFAULT_COLUMN_FLEX);
+      state = focusEditor(
+        action.payload.componentName.replace('editor.', ''),
```

Comment:
This is definitely the most low-impact way to do what we need to do here, but I’m not in love with the idea of parsing a string that’s just being used as an internal identifier. Seems like it might be better to change the structure of the action a bit so that the information we need is discretely available. Maybe something like `{componentType: 'editor', componentId: 'javascript'}`?

Reply:
Okay! That makes a lot of sense! Thanks :)